iam/oauth.go

```diff
@@ -195,37 +177,23 @@ func GetKeyvaultToken(grantType OAuthGrantType) (authorizer autorest.Authorizer,
 
 	switch grantType {
 	case OAuthGrantTypeServicePrincipal:
-		spt, err := adal.NewServicePrincipalToken(
-			*config,
-			clientID,
-			clientSecret,
-			"https://vault.azure.net")
-
+		token, err := adal.NewServicePrincipalToken(*config, clientID, clientSecret, vaultEndpoint)
```

Comment:
why isn't this one using the same auth mechanism ?

Reply:
It needs a different token endpoint